source/common/router/config_impl.cc

```diff
@@ -567,6 +571,14 @@ void RouteEntryImplBase::finalizeRequestHeaders(Http::RequestHeaderMap& headers,
     request_headers_parser_->evaluateHeaders(headers, stream_info);
   }
 
+  // Restore the port if this was a CONNECT request.
+  if (stream_info.filterState().hasData<OriginalConnectPort>(OriginalConnectPort::key()) &&
+      headers.getHostValue().find(':') == absl::string_view::npos) {
```

Comment:
qq is this sufficient to check for ports given that ipv6 addresses can have : in them? Or do we normalize that away somewhere?